### PR TITLE
#166214096 User receive Mail upon successful Registration

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?
Send Mail to the user upon successful Registration
#### Description of Task to be completed?
document API using swagger
#### How should this be manually tested?
```
clone or download this repository
npm install
create an .env file
add the ADMIN_MAIL in the .env file
and sendGrid_API_KEY in the .env file
in postman sign up
signUp mail will be delivered to the mail you registered with
for the first-time mail, may fall in the spam, go to spam directory and click on it to remove it from entering the spam again
```
#### What are the relevant pivotal tracker stories?
#166214096
#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by Lundii — COMMENTED** [review]
You might need to send us the ADMIN_MAIL and SENDGRID_API_KEY privately so we don't have create new mails in other to test this PR.

sent to all reviewers Direct Message on slack

**Review by Lundii — COMMENTED** [review]
I'm sure its working on your local, however I cant get it to work on my local, the error is coming from the send grid key. I used the parameters you sent to me.
 This is the error
Screenshot (62)

When I signup, I get this
Screenshot (60)

When I tried to try again, I get this
Screenshot (61)


ENSURE YOUR text FIELDS AND html fields in your "messages" are not empty first
you're getting this error because you are inputting something that is not valid
if it was the secret key then you'll get an unAuthorized

**Comment by allebd** [comment]
Kindly rebase and I observed, all your spelling of **successful** in your PR title, your PR description are wrong

Treated accordingly  thanks

File: server/controllers/userController.js [hunk 1]
```diff
@@ -16,20 +16,29 @@ const { User } = models;
...
       if (existingUser) {
         return responseMessage(response, 409, { message: 'user already exist' });
       }
       const newUser = await User.create({ firstName, lastName, email, password: bcrypt.hashSync(password, 10), avatarUrl, verified: false })
       if (newUser) {
         const { id } = newUser.dataValues
         const token = createToken({ id }, '24h');
+    const message = {
+       text: 'hello Books',
+    html: '<strong> Welcome to hello books (SIGN UP SUCESSFULL)</div>'
```
Comment (allebd) [inline server/controllers/userController.js]: Wrong spelling of successful

**Comment by allebd** [comment]
Have tested it, it is working fine, all that is left is to rebase and correct **successful** in 3 places above

File: server/controllers/userController.js (hunk 1 above)
Comment (Adekoreday) [reply server/controllers/userController.js]: done

**Review by allebd — DISMISSED** [review]
Well done

File: .env.sample [hunk 2]
```diff
@@ -9,3 +9,5 @@ DATABASE_TEST=
 DATABASE_URL=
 JWT_KEY=
 SENDGRID_API_KEY=
+ADMIN_MAIL
+VERIFY_TOKEN_URL
```
Comment (henryade) [inline .env.sample]: add new line

File: bash.exe.stackdump [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Stack trace:
+Frame        Function    Args
+000005FA218  0018007208E (0018026056D, 00180215E46, 000005FA218, 000005F9110)
+000005FA218  00180046DF2 (000005FA178, 00000000000, 00000000000, 000FFFFFFFF)
+000005FA218  00180046E32 (00180260629, 000005FA0C8, 000005FA218, 00000000000)
+000005FA218  001800BDE7F (00000000000, 00000000000, 00000000000, 00000000000)
+000005FA218  001800BE09F (000005FA240, 00000000000, 00000000000, 00000000000)
+000005FA2C0  001800BF35A (000005FA240, 00000000000, 00000000000, 00000000000)
+End of stack trace
```
Comment (henryade) [inline bash.exe.stackdump]: You havent removed this file

Comment (Adekoreday) [reply bash.exe.stackdump]: ok thanks will look into it

**Review by JuwonAbiola — DISMISSED** [review]

**Pull request merged**